File: PHP/DesignPattern/FactoryMethod/NYPizzaStore.php
<?php

	include 'PizzaStore.php';
	include 'NYStyleCheesePizza.php';
	include 'NYStyleVegglePizza.php';

	/**
	* 具体创建者类
	* 继承自工厂方法类, 能产生产品的类
	*/
	class NYPizzaStore extends PizzaStore {

		/**
		* 实现抽象方法createPizza，产生具体产品
		*/
		public function createPizza($type) {
			if ($type == "cheese") {
				return new NYStyleCheesePizza();
			} else if ($type == "veggle") {
				return new NYStyleVegglePizza();
			} else {
				return null;
			}
		}
	}

File: Java/FactoryMethod/src/com/base/NYPizzaStore.java
package com.base;

import com.base.Pizza;
import com.base.PizzaStore;

/**
 * Created by hhq on 1/14/15.
 */
public class NYPizzaStore extends PizzaStore {
    protected Pizza createPizza(String item) {
        if (item.equals("cheese")) {
            return new NYStyleCheesePizza();
        } else if (item.equals("veggie")) {
            return new NYStyleVeggiePizza();
        } else if (item.equals("clam")) {
            return new NYStyleClamPizza();
        } else if (item.equals("pepperoni")) {
            return new NYStylePepperoni();
        } else 
            return null;
    }
}
